jspwiki-main/src/test/java/org/apache/wiki/auth/DefaultUserManagerTest.java

```diff
@@ -0,0 +1,795 @@
+import org.apache.wiki.api.core.Command;
+import org.apache.wiki.api.core.Context;
+import org.apache.wiki.api.core.Engine;
+import org.apache.wiki.api.core.Page;
+import org.apache.wiki.api.core.Session;
+import org.apache.wiki.api.exceptions.WikiException;
+import org.apache.wiki.auth.user.UserProfile;
+import org.apache.wiki.event.WikiEvent;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.security.auth.Subject;
+import javax.servlet.AsyncContext;
+import javax.servlet.DispatcherType;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletInputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import javax.servlet.http.HttpSessionContext;
+import javax.servlet.http.HttpUpgradeHandler;
+import javax.servlet.http.Part;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.security.Permission;
+import java.security.Principal;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+class DefaultUserManagerTrimTest {
+
+    private Engine engine;
+    private DefaultUserManager userManager;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        // Initialize a minimal test engine.
+        engine = new TestEngine();
+        // Get the UserManager (which should be an instance of DefaultUserManager).
+        userManager = ( DefaultUserManager ) engine.getManager( UserManager.class );
+    }
+
+    @Test
+    public void testParseProfileTrimsFields() throws WikiException {
+        // Create a dummy HttpServletRequest that returns parameters with extra whitespace.
+        HttpServletRequest request = new DummyHttpServletRequest();
+        // Create a dummy Context that returns our dummy request and a dummy Session.
+        Context dummyContext = new DummyContext( engine, request );
+
+        // Call parseProfile, which should trim the input values.
+        UserProfile profile = userManager.parseProfile( dummyContext );
+
+        // Verify that the login name, fullname, and email have been trimmed.
+        Assertions.assertEquals( "admin", profile.getLoginName(), "Login name should be trimmed" );
+        Assertions.assertEquals( "Administrator", profile.getFullname(), "Full name should be trimmed" );
+        Assertions.assertEquals( "admin@example.com", profile.getEmail(), "Email should be trimmed" );
+    }
+
+    // --- Minimal dummy implementations ---
+
+    // A very basic HttpServletRequest implementation supporting getParameter.
+    private static class DummyHttpServletRequest implements HttpServletRequest {
```

Comment:
Both `DummyHttpServletRequest` and `DummyContext` should better be mocked using mockito